src/natcap/invest/managed_raster/ManagedRaster.h

```diff
@@ -0,0 +1,652 @@
+                return;
+            }
+            closed = 1;
+
+            double *double_buffer;
+            int block_xi;
+            int block_yi;
+            int block_index;
+            // initially the win size is the same as the block size unless
+            // we're at the edge of a raster
+            int win_xsize;
+            int win_ysize;
+
+            // we need the offsets to subtract from global indexes for cached array
+            int xoff;
+            int yoff;
+
+            if (not write_mode) {
+                for (auto it = lru_cache->begin(); it != lru_cache->end(); it++) {
+                    // write the changed value back if desired
+                    CPLFree(it->second);
+                }
+                return;
+            }
+
+            // if we get here, we're in write_mode
+            std::set<int>::iterator dirty_itr;
+            for (auto it = lru_cache->begin(); it != lru_cache->end(); it++) {
+                double_buffer = it->second;
+                block_index = it->first;
+
+                // write to disk if block is dirty
+                dirty_itr = dirty_blocks.find(block_index);
+                if (dirty_itr != dirty_blocks.end()) {
+                    dirty_blocks.erase(dirty_itr);
+                    block_xi = block_index % block_nx;
+                    block_yi = block_index / block_nx;
+
+                    // we need the offsets to subtract from global indexes for
+                    // cached array
+                    xoff = block_xi << block_xbits;
+                    yoff = block_yi << block_ybits;
+
+                    win_xsize = block_xsize;
+                    win_ysize = block_ysize;
+
+                    // clip window sizes if necessary
+                    if (xoff + win_xsize > raster_x_size) {
+                        win_xsize = win_xsize - (xoff + win_xsize - raster_x_size);
+                    }
+                    if (yoff + win_ysize > raster_y_size) {
+                        win_ysize = win_ysize - (yoff + win_ysize - raster_y_size);
+                    }
+                    CPLErr err = band->RasterIO( GF_Write, xoff, yoff, win_xsize, win_ysize,
+                        double_buffer, win_xsize, win_ysize, GDT_Float64, 0, 0 );
+                    if (err != CE_None) {
+                        std::cerr << "Error writing block\n";
+                    }
+                }
+                CPLFree(double_buffer);
+            }
+            GDALClose( (GDALDatasetH) dataset );
+            delete lru_cache;
+            free(actualBlockWidths);
+        }
+};
+
+class ManagedFlowDirRaster: public ManagedRaster {
+
+public:
+
+    ManagedFlowDirRaster() {}
+
+    ManagedFlowDirRaster(char* raster_path, int band_id, bool write_mode)
+        : ManagedRaster(raster_path, band_id, write_mode)   // Call the superclass constructor in the subclass' initialization list.
+        {
+            // do something with bar
+        }
```

Comment:
Is the plan to still do something in this constructor?

Reply:
Fixed in 25c954af